src/Compilers/CSharp/Test/Syntax/Parsing/TargetTypedNewParsingTests.cs

```diff
@@ -0,0 +1,469 @@
+        [InlineData(SyntaxKind.ModuloExpression, SyntaxKind.PercentToken)]
+        [InlineData(SyntaxKind.LeftShiftExpression, SyntaxKind.LessThanLessThanToken)]
+        [InlineData(SyntaxKind.RightShiftExpression, SyntaxKind.GreaterThanGreaterThanToken)]
+        [InlineData(SyntaxKind.LogicalOrExpression, SyntaxKind.BarBarToken)]
+        [InlineData(SyntaxKind.LogicalAndExpression, SyntaxKind.AmpersandAmpersandToken)]
+        [InlineData(SyntaxKind.BitwiseOrExpression, SyntaxKind.BarToken)]
+        [InlineData(SyntaxKind.BitwiseAndExpression, SyntaxKind.AmpersandToken)]
+        [InlineData(SyntaxKind.ExclusiveOrExpression, SyntaxKind.CaretToken)]
+        [InlineData(SyntaxKind.EqualsExpression, SyntaxKind.EqualsEqualsToken)]
+        [InlineData(SyntaxKind.NotEqualsExpression, SyntaxKind.ExclamationEqualsToken)]
+        [InlineData(SyntaxKind.LessThanExpression, SyntaxKind.LessThanToken)]
+        [InlineData(SyntaxKind.LessThanOrEqualExpression, SyntaxKind.LessThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanExpression, SyntaxKind.GreaterThanToken)]
+        [InlineData(SyntaxKind.GreaterThanOrEqualExpression, SyntaxKind.GreaterThanEqualsToken)]
+        public void TestBinaryOperators(SyntaxKind expressionKind, SyntaxKind tokenKind)
+        {
+            UsingExpression($"new(Int32,Int32){SyntaxFacts.GetText(tokenKind),2}",
+                // (1,18): error CS1733: Expected expression
+                // new(Int32,Int32) +
+                Diagnostic(ErrorCode.ERR_ExpressionExpected, "").WithLocation(1, 19));
+
+            UsingExpression($"new(Int32,Int32){SyntaxFacts.GetText(tokenKind),2}e");
+
+            N(expressionKind);
+            {
+                N(SyntaxKind.ObjectCreationExpression);
+                {
+                    N(SyntaxKind.NewKeyword);
+                    N(SyntaxKind.ArgumentList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.Argument);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "Int32");
+                            }
+                        }
+                        N(SyntaxKind.CommaToken);
+                        N(SyntaxKind.Argument);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "Int32");
+                            }
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                }
+                N(tokenKind);
+                N(SyntaxKind.IdentifierName, "e");
+                {
+                    N(SyntaxKind.IdentifierToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestEmptyArgList()
```

Comment:
Please add a variant of this test with LangVersion 7.3 (I expect diagnostics). Same for two tests below (`TestEmptyObjectInitializer` and `TestObjectInitializer`).